File: 2825-make-string-a-subsequence-using-cyclic-increments/2825-make-string-a-subsequence-using-cyclic-increments.java
class Solution {
    public boolean canMakeSubsequence(String str1, String str2) {
        int j = 0;
        for (char c : str1.toCharArray()) {
            if (j < str2.length() && (c == str2.charAt(j) || c - str2.charAt(j) == 25 || c - str2.charAt(j) == -1)) {
                j++;
            }
        }
        return j == str2.length();
    }
}

File: 2825-make-string-a-subsequence-using-cyclic-increments/2825-make-string-a-subsequence-using-cyclic-increments.py
class Solution:
    def canMakeSubsequence(self, str1: str, str2: str) -> bool:
        j = 0
        for c in str1:
            if j < len(str2) and (c == str2[j] or (ord(c) - ord(str2[j]) == 25) or (ord(c) - ord(str2[j]) == -1)):
                j += 1
        return j == len(str2)
